### PR TITLE
replace `run_cmd` with `run_shell_cmd` in custom easyblock for Perl (`perl.py`)

### PR DESCRIPTION
**Pull request by bartoldeman**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Perl-5.32.0-GCCcore-10.2.0.eb_ 
 * **SUCCESS** _Perl-5.38.0-GCCcore-13.2.0.eb_ 
 * **SUCCESS** _Perl-5.34.1-GCCcore-11.3.0-minimal.eb_ 
 * **SUCCESS** _Perl-5.34.0-GCCcore-11.2.0.eb_ 

Build succeeded for 4 out of 4 (4 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/f8dfb469c1d38406c0a9729b0f9d1b02 for a full test report.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @bartoldeman!

File: easybuild/easyblocks/p/perl.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def test_step(self):
             # specify locale to be used, to avoid that a handful of tests fail
             cmd = "export LC_ALL=C && %s" % cmd
 
-            run_cmd(cmd, log_all=False, log_ok=False, simple=False)
+            run_shell_cmd(cmd, fail_on_error=False)
```
Comment (boegel) [inline easybuild/easyblocks/p/perl.py:142]: Why are we ignoring non-zero exit codes here? That seems wrong...

@bartoldeman I understand that you're just translating from `run_cmd` to `run_shell_cmd` without changing behavior here, but this looks like a bug we should fix, no?

Comment (branfosj) [reply easybuild/easyblocks/p/perl.py:142]: see #3170

**Pull request merged**